### PR TITLE
DEV-2963/DEV-2853: Documentation fixes

### PR DESCRIPTION
**Pull request by kbard**
**Description:**

Adds and fixes endpoint summary links and fixes/adds endpoint specific documentation.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [x] Necessary PR reviewers:
    - [x] Backend
4. [x] Materialized views unaffected
5. [x] Frontend unaffected
6. [x] No data affected
7. [x] No Operations tickets required
8. [x] Jira Ticket [DEV-2963](https://federal-spending-transparency.atlassian.net/browse/DEV-2963)/[DEV-2853](https://federal-spending-transparency.atlassian.net/browse/DEV-2853):
    - [x] Link to this Pull-Request

### CODE REVIEW

File: usaspending_api/accounts/urls_federal_accounts_v2.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 urlpatterns = [
     url(r"(?P<pk>[0-9]+)/available_object_classes$", object_class_federal_accounts),
     url(r"(?P<pk>[0-9]+)/fiscal_year_snapshot/(?P<fy>[0-9]+)/?$", fiscal_year_snapshot_federal_accounts),
-    url(r"(?P<pk>[0-9]+)/fiscal_year_snapshot$", fiscal_year_snapshot_federal_accounts),
+    url(r"(?P<pk>[0-9]+)/fiscal_year_snapshot/?$", fiscal_year_snapshot_federal_accounts),
```
Comment (BrianZito) [inline usaspending_api/accounts/urls_federal_accounts_v2.py]: Why was this added? Should we be adding this for everything?

Comment (kbard) [reply usaspending_api/accounts/urls_federal_accounts_v2.py]: It was the only endpoint that didn't work with a slash on the end, so I added support.

File: usaspending_api/common/renderers.py [hunk 2]
```diff
@@ -62,100 +67,181 @@ class GenericContentForm(forms.Form):
...
+        we can find a file that meets our needs.
+        """
+        if doc:
+            file = os.path.split(doc)[1]
+            for location in SEARCH_LOCATIONS:
+                # Don't attempt to check a location where a version
+                # is required but we don't have one.
+                if "{version}" not in location or version:
+                    doc_path = os.path.join(BASE_DIR, location.format(version=version), doc)
+                    # This is stupid.  This is a workaround for case insensitive file systems.
```
Comment (BrianZito) [inline usaspending_api/common/renderers.py]: perfection

File: usaspending_api/common/management/commands/check_for_endpoint_documentation.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        messages = []
+        for endpoint in endpoints:
+            messages.extend(validate_docs(*endpoint, master_endpoint_list))
+
+        if messages:
+            for message in messages:
+                print(message)
+            exit(1)
+
+        print("Looks like endpoint documentation is happy, healthy, wealthy, and wise")
```
Comment (tony-sappe) [inline usaspending_api/common/management/commands/check_for_endpoint_documentation.py]: Our API docs definitely aren't happy, nor wealthy

File: usaspending_api/awards/v2/views/award_spending.py [hunk 4]
```diff
@@ -15,7 +15,6 @@
 class AwardTypeAwardSpendingViewSet(CachedDetailViewSet):
     """
     Return all award spending by award type for a given fiscal year and agency id
-    endpoint_doc: award_spending/award_category.md
```
Comment (tony-sappe) [inline usaspending_api/awards/v2/views/award_spending.py]: I'm confused why this endpoint doesn't need a doc

File: usaspending_api/common/management/commands/check_for_endpoint_documentation.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    def handle(self, *args, **options):
+        master_endpoint_list = get_endpoints_from_endpoints_markdown()
+
+        endpoints = get_endpoint_urls_doc_paths_and_docstrings(CURRENT_ENDPOINT_PREFIXES)
+
+        messages = []
+        for endpoint in endpoints:
+            messages.extend(validate_docs(*endpoint, master_endpoint_list))
+
```
Comment (tony-sappe) [inline usaspending_api/common/management/commands/check_for_endpoint_documentation.py]: Minor request: Can you add a check to count or list the endpoints which have a document however the doc isn't an API contract? This shouldn't make the check not pass. But it would be useful to track the v2 endpoints without a contract.

File: usaspending_api/common/helpers/endpoint_documentation.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+import re
+
+from usaspending_api.settings import BASE_DIR
+
+
+# Typically we only care about v2 API endpoints, but if we ever add v3 or
+# whatever, add the base path to this tuple.
+CURRENT_ENDPOINT_PREFIXES = ("/api/v2/",)
+
+ENDPOINTS_MD = "usaspending_api/api_docs/markdown/endpoints.md"
```
Comment (tony-sappe) [inline usaspending_api/common/helpers/endpoint_documentation.py]: What about API Contracts? We are moving toward exclusively using them for api documents

File: usaspending_api/common/helpers/endpoint_documentation.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    for endpoint in master_endpoint_list:
+        if re.fullmatch(pattern, endpoint):
+            break
+    else:
+        messages.append("No URL found in {} that matches {} ({})".format(ENDPOINTS_MD, url, qualified_name))
+
+    return messages
+
+
+__all__ = [
```
Comment (tony-sappe) [inline usaspending_api/common/helpers/endpoint_documentation.py]: What determined the reason to keep two of the items out of __all__ ? (If I'm interpreting what is the feature of explicitly setting `__all__`

File: usaspending_api/awards/v2/views/award_spending.py (hunk 4 above)
Comment (kbard) [reply usaspending_api/awards/v2/views/award_spending.py]: It's not being used anywhere.

File: usaspending_api/common/helpers/endpoint_documentation.py (hunk 6 above)
Comment (kbard) [reply usaspending_api/common/helpers/endpoint_documentation.py]: I don't understand what you are asking.

File: usaspending_api/common/helpers/endpoint_documentation.py (hunk 7 above)
Comment (kbard) [reply usaspending_api/common/helpers/endpoint_documentation.py]: They weren't being used anywhere outside of the module.  Honestly, I don't know why I even bothered with the `__all__`.

File: usaspending_api/common/helpers/endpoint_documentation.py (hunk 6 above)
Comment (tony-sappe) [reply usaspending_api/common/helpers/endpoint_documentation.py]: that makes two of us....
